lib/spack/spack/url.py

```diff
@@ -199,6 +201,10 @@ def parse_version_offset(path):
         # https://github.com/hpc/mpileaks/releases/download/v1.0/mpileaks-1.0.tar.gz
         (r'github.com/[^/]+/[^/]+/releases/download/v?([^/]+)/.*$', path),
 
+        # e.g., https://gitlab.kitware.com/vtk/vtk/repository/archive.tar.bz2?ref=v7.0.0
+        # e.g., https://gitlab.kitware.com/icet/icet/repository/archive.tar.bz2?ref=IceT-2.1.1
```

Comment:
Thanks for the questions.

> right. I would then mention it here as well, saying something like "here we assume that git tag ends with a dotted version x.y.z".

Good point, will add it to the [`split_url_extension` docstring](https://github.com/ax3l/spack/blob/d40af726dae0e7867a7ed65707d22b00055802cc/lib/spack/spack/url.py#L107-L109) as we do for the GitHub repos.

> also since it is specific to gitlab, should not your regex start with 'gitlab.?
> on a second thought [...]

Unfortunately no, I documented that in the PR description. There are many gitlab installs out there which are not installed in a specific sub-domain that reads `gitlab.` :)